### PR TITLE
feat: add welcome to the jungle scraper tool

### PR DESCRIPTION
**Pull request by Smirkey**

### CODE REVIEW

File: aiden_app/tools/tool.py [hunk 1]
```diff
@@ -11,7 +12,7 @@ def __init__(self, name: str):
     def add_tool(self, name, function, agent_speaks_next=True):
         self.names_to_functions[name] = functools.partial(function, func_name=name, agent_speaks_next=agent_speaks_next)
 
-    def tool_function(func):
+    def tool_function(func: Self):
```
Comment (Anko59) [inline aiden_app/tools/tool.py:15]: You sure func should be Self? My assumption would be Callable

File: aiden_app/tools/utils/wtj_scraper.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        self.driver = ChromeDriver()
+
+    def _fetch_results(self, search_query: str, location: str):
+        headers = {
+            "User-Agent": "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:123.0) Gecko/20100101 Firefox/123.0",
+            "Accept": "*/*",
+            "Accept-Language": "en-US,en;q=0.5",
+            "Referer": "https://www.welcometothejungle.com/",
+            "x-algolia-api-key": "4bd8f6215d0cc52b26430765769e65a0",
+            "x-algolia-application-id": "CSEKHVMS53",
```
Comment (Smirkey) [inline aiden_app/tools/utils/wtj_scraper.py]: we could use selenium to refresh this, don't know if it's necessary though

File: aiden_app/tools/utils/wtj_scraper.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    def to_url(self) -> str:
+        return f"https://www.welcometothejungle.com/fr/companies/{self.organization.name.lower()}/jobs/{self.slug}?&o={self.reference}"
+
+
+class WelcomeToTheJungleScraper:
+    def __init__(self):
+        self.driver = ChromeDriver()
+
+    def _fetch_results(self, search_query: str, location: str):
+        headers = {
```
Comment (Anko59) [inline aiden_app/tools/utils/wtj_scraper.py]: Can we shorten the headers here?

File: aiden_app/tools/wtj_scraper_tool.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+class WelcomeToTheJungleScraperTool(Tool):
+    def __init__(self):
+        self.scraper = WelcomeToTheJungleScraper()
+        super().__init__("IndeedScraper")
+        self.add_tool("search_jobs", self.search_jobs_wrapper)
+
+    @Tool.tool_function
+    def search_jobs_wrapper(self, search_query: str, location: str, num_results: int = 15, start: int = 0):
+        results = self.scraper.search_jobs(search_query, location, int(num_results), int(start))
+        return results
```
Comment (Smirkey) [inline aiden_app/tools/wtj_scraper_tool.py]: this is not satisfying nor ready to be merged, the tool will simply return its own data format which has nothing to do with indeed's
furthermore, we need to de duplicate offers maybe

File: aiden_app/tools/wtj_scraper_tool.py [hunk 5]
```diff
@@ -0,0 +1,14 @@
+from .tool import Tool
+from .utils.wtj_scraper import WelcomeToTheJungleScraper
+
+
+class WelcomeToTheJungleScraperTool(Tool):
+    def __init__(self):
+        self.scraper = WelcomeToTheJungleScraper()
+        super().__init__("IndeedScraper")
```
Comment (Anko59) [inline aiden_app/tools/wtj_scraper_tool.py]: Wrong name

File: aiden_app/tools/wtj_scraper_tool.py [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+        self.add_tool("search_jobs", self.search_jobs_wrapper)
```
Comment (Anko59) [inline aiden_app/tools/wtj_scraper_tool.py:9]: I don't see how this is gonna work. In the `ToolAggregator`, the last function with the same name (search_jobs) will overwrite the previous one.

File: aiden_app/tools/tools.json [hunk 7]
```diff
@@ -47,7 +47,7 @@
         "type": "function",
         "function": {
             "name": "search_jobs",
-            "description": "Search the website Indeed for job offers. Ranked by newest first.",
+            "description": "Search the the internet for job offers. Ranked by newest first.",
```
Comment (Anko59) [inline aiden_app/tools/tools.json:50]: We either need two separate tools for indeed and wttj scapers, or one unified tool with additional engineering .

**Review by Anko59 — CHANGES_REQUESTED** [review]
Main issue here is that Wtj scraper overwrites Indeed scraper

File: aiden_app/tools/utils/wtj_scraper.py [hunk 8]
```diff
@@ -89,24 +131,29 @@ def __init__(self):
             "apiKey": script_dict["HERE_API_KEY"],
             "lang": "fr",
         }
+        logger.info("succesfully initialized WTJ scraper")
 
-    def _get_algolia_params(self, search_query: str, latlng: str):
-        params = {"hitsPerPage": 300, "query": search_query, "aroundLatLng": latlng, "aroundRadius": 2000000}
+    def _get_algolia_params(self, search_query: str, latlng: str, page: int):
+        params = {"hitsPerPage": 1000, "query": search_query, "aroundLatLng": latlng, "aroundRadius": 200000000000000, "page": page}
```
Comment (Anko59) [inline aiden_app/tools/utils/wtj_scraper.py]: Are the results ranked by location ? Does the location even matter with such a large radius?

**Review by Anko59 — APPROVED** [review]

**Comment by Smirkey** [comment]
Let's switch to #12

**Pull request closed without merging**